src/alda/lisp/model/key.clj

```diff
@@ -54,8 +54,13 @@
   [scale-type]
   (zipmap (map (comp keyword str) "fcgdaeb")
           (case scale-type
-            :major (range -1 6)
-            :minor (range -4 3))))
+            (:major :ionian) (range -1 6)
+            (:minor :aeolian) (range -4 3)
```

Comment:
This will not work, unfortunately: Clojure interprets`(:major :ionian)` as "call `:major` like it's a function on the argument `:ionian`.

In Clojure, keywords implement the IFn protocol, which allows them to act as functions that look themselves up in a map. For example:

```clojure
boot.user=> (:b {:a 1 :b 2 :c 3})
2
```

If the argument is not a map, however, the result can be `nil`:

```clojure
boot.user=> (:major :ionian)
nil
```

So I think the major and minor modes are silently failing here because of that.

What we could use here is [condp](https://clojuredocs.org/clojure.core/condp) using `contains?` as the predicate and sets of accepted values as cases:

```clojure
boot.user=> (condp contains? 5
       #_=>   #{1 2 3 4} "does not contain 5"
       #_=>   #{5 6 7}   "contains 5"
       #_=>   #{8 9 10}  "also does not contain 5")
"contains 5"

boot.user=> (condp contains? :aeolian
       #_=>   #{:major :ionian}  (range -1 6)
       #_=>   #{:minor :aeolian} (range -4 3)
       #_=>   #{:lydian}         (range 0 7))
(-4 -3 -2 -1 0 1 2)
```

Reply:
EDIT: never mind! see comments below :)